presto-orc/src/test/java/com/facebook/presto/orc/TestSelectiveOrcReader.java

```diff
@@ -580,6 +580,18 @@ public void testArraysOfNulls()
         }
     }
 
+    @Test
+    public void testArraysOfNullsWithNullFilterAndOtherColumnFilters()
```

Comment:
@bhhari 

I understand that the issue is that the return value of `getReadPositions()` is junk and sometimes that's ok while sometimes it is not. For example, boolean reader can read positions [0, 0, 0, 0, 0] as if they are [0, 1, 2, 3, 4]). It might be useful to add a sanity check to catch such corruptions earlier. For example, we could add the following to OrcSelectiveRecordReader:

```
            positionsToRead = streamReader.getReadPositions();

            verify(positionCount == 1 || positionsToRead[positionCount - 1] - positionsToRead[0] >= positionCount - 1, "positions must monotonically increase");
```

or to each reader's read method:

```
checkArgument(positionCount == 1 || positions[positionCount - 1] - positions[0] >= positionCount - 1, "positions must monotonically increase");
```

My preference is for the former because it is less code.

Once we do that, we can catch these issues much more easily, e.g. a test like the following would be sufficient.

```
    @Test
    public void testArraysOfNullsWithFilter()
            throws Exception
    {
        tester.testRoundTripTypes(ImmutableList.of(arrayType(BIGINT), BOOLEAN),
                ImmutableList.of(nCopies(5, null), newArrayList(true, false, null, true, false)),
                ImmutableList.of(ImmutableMap.of(0, ImmutableMap.of(new Subfield("c"), IS_NULL))));
    }
```